### PR TITLE
[WIP] tech extension: update backend to support extensions #725

### PR DESCRIPTION
**Pull request by lburgazzoli**
# WIP - DO NOT MERGE

I've amended the model according to #742 (but without taking extension into account for project generation and activation handler) and this PR is for collecting feedback and reviews.

@syndesisio/backend could you please carefully review it ?
@gashcrumb the connector information (gav, prefix, id) has been mode to the ConnectorDescriptor as per discussion on #742 would you mind having a look too ?

### CODE REVIEW

**Comment by lburgazzoli** [comment]
A relevant topic for this PR is the change in Action which is now a base interface for ConnectorAction and ExtensionAction:

```java
public interface Action<D extends ActionDescriptor> 
    extends WithName, WithTags, WithConfigurationProperties, Serializable {
    ...
}
```

```java
public interface ConnectorAction 
    extends Action<ConnectorDescriptor>, WithId<ConnectorAction> {
    
    @Override
    default Kind getKind() {
        return Kind.ConnectorAction;
    }
}
```
```java
public interface ExtensionAction 
    extends Action<ExtensionDescriptor>, WithId<ExtensionAction> {

    @Override
    default Kind getKind() {
        return Kind.ExtensionAction;
    }
}
```

This means that there's no more `Kind.Action` but we have now:

- Kind.ConnectorAction
- Kind.ExtensionAction

I do not know how deep this impacts.

**Comment by lburgazzoli** [comment]
retest this please

**Comment by lburgazzoli** [comment]
retest this please

**Comment by rhuss** [comment]
@lburgazzoli sorry, I only could have a super short look but plan to dedicate much more time when I'm back from w-jax (friday). @jimmidyson @KurtStam @zregvart @iocanel @chirino  If time permits it would be awesome to get a review, too, as this is part of some more fundamental change in the model.

**Comment by lburgazzoli** [comment]
retest this please

**Comment by lburgazzoli** [comment]
retest this please

**Comment by lburgazzoli** [comment]
retest this please

**Comment by lburgazzoli** [comment]
retest this please

**Comment by lburgazzoli** [comment]
retest this please

File: dao/src/test/resources/logback.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<!--
```
Comment (zregvart) [inline dao/src/test/resources/logback.xml]: perhaps name this file `logback-test.xml`, to help poor souls that run from unpopular IDE

**Comment by lburgazzoli** [comment]
retest this please

**Pull request closed without merging**